feat(cicd): add conditional field in pipeline manifest for connection_name (#2073)
aws/copilot-cli: site/content/docs/manifest/pipeline.md

      properties:
        branch: main
        repository: https://github.com/<user>/sample-app-frontend
        # Optional: specify the name of an existing CodeStar Connections connection.

>         connection_name:

Reviewer: Should there be an example connection name here?

Author: Added. Thanks!

>         connection_name: MyPreexistingConnection

Reviewer 2: nit: samesies here

<span class="parent-field">source.properties.</span><a id="source-properties-repository" href="#source-properties-repository" class="field">`repository`</a> <span class="type">String</span>  
The URL of your repository.

<span class="parent-field">source.properties.</span><a id="source-properties-connection-name" href="#source-properties-connection-name" class="field">`connection_name`</a> <span class="type">String</span>  

> The name of an existing CodeStar Connections connection. By default, Copilot will generate a connection for you.

Reviewer 2: ```suggestion
The name of an existing CodeStar Connections connection. If omitted, Copilot will generate a connection for you.
```

Author: Ah, that's much better. Thanks!
